### PR TITLE
Bugfix | Filter out not enabled products in API collection

### PR DESCRIPTION
**Pull request by stloyd**
| Q               | A
| --------------- | -----
| Branch?         | 1.8
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Right now API `/shop/products` returns all products in database, even those that are explicitly disabled.

### CODE REVIEW

**Review by pamil — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/ProductsWithEnableFlagExtension.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/ProductsWithEnableFlagExtension.php:2]: missing license block

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryCollectionExtension/ProductsWithEnableFlagExtensionSpec.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+use Doctrine\ORM\QueryBuilder;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class ProductsWithEnableFlagExtensionSpec extends ObjectBehavior
```
Comment (Zales0123) [inline src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryCollectionExtension/ProductsWithEnableFlagExtensionSpec.php:26]: Missing `final` 🖖

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, Joseph! :1st_place_medal:

**Comment by stloyd** [comment]
Thanks to @arti0090 for finishing it up 🙇‍♂️

**Pull request merged**